c10/xpu/XPUException.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <c10/util/Exception.h>
+#include <sycl/sycl.hpp>
+
+namespace c10::xpu {
+
+static inline sycl::async_handler asyncHandler = [](sycl::exception_list el) {
+  if (el.size() == 0) {
+    return;
+  }
+  for (const auto& e : el) {
+    try {
+      std::rethrow_exception(e);
+    } catch (sycl::exception& e) {
+      TORCH_WARN("SYCL Exception: ", e.what());
+    }
+  }
+  throw;
```

Comment:
Why warn above and then throw with no context?
You can use `TORCH_CHECK(false, "Some error with details: ", e.what());` to throw an error that will become a python RuntimeError.
We also have variants for all python error types like `TORCH_CHECK_TYPE` for TypeError or `TORCH_CHECK_NOT_IMPLEMENTED` for NotImplementedError.

Reply:
We would like to show all sycl exceptions contained in the sycl exception_list. So we warn each sycl exception in line 16 and terminate the program in line 19. Sorry for confusing you, we will submit an additional PR to refine here.

Reply:
Ho that makes sense thanks for clarifying.
I think warning for all but the last one is a good way to go then? And the last one is a real error.

Reply:
Thanks for your input.